src/main/java/org/jitsi/videobridge/LipSyncHack.java

```diff
@@ -0,0 +1,423 @@
+                int seqnumDelta = RTPUtils.subtractNumber(
+                    lastSeqnumDropped, state.numOfKeyframesSent);
+
+                long timestamp
+                    = RawPacket.getTimestamp(buffer, offset, length);
+
+                // Timestamps are calculated.
+                long highestTimestampSent = state.numOfKeyframesSent * 3000;
+
+                // Pretend we have dropped all the packets prior to the one
+                // that's about to be written by the translator.
+                long lastTimestampDropped = (timestamp - 3000) & 0xffffffffl;
+                long timestampDelta
+                    = (lastTimestampDropped - highestTimestampSent) & 0xffffffff;
+
+                rewriter = new ResumableStreamRewriter(
+                    state.numOfKeyframesSent, seqnumDelta,
+                    highestTimestampSent, timestampDelta);
+
+                streamRTPManager.ssrcToRewriter.put(acceptedVideoSSRC, rewriter);
+            }
+            else
+            {
+                logger.warn("Could not initialize a sequence number rewriter" +
+                    "because one is already there.");
+            }
+        }
+
+    }
+
+    /**
+     *
+     */
+    class InjectTask implements Runnable
+    {
+        /**
+         * The state for this injector.
+         */
+        private final InjectState injectState;
+
+        /**
+         * The {@link ScheduledFuture} for this task.
+         */
+        private ScheduledFuture<?> scheduledFuture;
+
+        /**
+         * Ctor.
+         *
+         * @param injectState
+         */
+        public InjectTask(InjectState injectState)
+        {
+            this.injectState = injectState;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void run()
+        {
+            synchronized (injectState)
+            {
+                if (!injectState.active
+                    || injectState.numOfKeyframesSent++ >= MAX_KEY_FRAMES)
+                {
+                    scheduledFuture.cancel(true);
+                    return;
+                }
+
+                MediaStream mediaStream = injectState.target.get();
+                if (mediaStream == null || !mediaStream.isStarted())
+                {
+                    logger.debug("Waiting for the media stream to become" +
+                        "available.");
+                    return;
+                }
+
+                try
+                {
+                    // FIXME maybe grab from the write pool and copy the array?
+                    byte[] buf = KEY_FRAME_BUFFER.clone();
+                    RawPacket keyframe = new RawPacket(buf, 0, buf.length);
+
+                    long timestamp = injectState.numOfKeyframesSent * 3000;
+                    keyframe.setSSRC(injectState.ssrc.intValue());
+                    keyframe.setSequenceNumber(
+                        injectState.numOfKeyframesSent);
```

Comment:
This will probably affect the sequence number space used for video packets, if video is enabled later. It should start with a randomly chosen value per the RTP specs.